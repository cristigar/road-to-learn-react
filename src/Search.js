import React from 'react';

const Search = ({
  searchTerm,
  onChange,
  children: text,
  onSubmit,
}) =>
  <form onSubmit={onSubmit}>
    <input
      type="search"
      onChange={onChange}
      value={searchTerm}
    />
    <button type="submit">{text}</button>
  </form>
;

export default Search;
