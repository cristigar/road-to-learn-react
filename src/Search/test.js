import Enzyme from "enzyme/build";
import Adapter from "enzyme-adapter-react-16";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import React from "react";

import Search from './index'

Enzyme.configure({ adapter: new Adapter() });

describe('Search', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Search>Search</Search>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
