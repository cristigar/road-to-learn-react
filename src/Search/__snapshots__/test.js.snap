// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={undefined}
>
  <input
    onChange={undefined}
    type="search"
    value={undefined}
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;
